ckan/lib/plugins.py

```diff
@@ -524,3 +526,38 @@ def activity_template(self):
         return 'organization/activity_stream.html'
 
 _default_organization_plugin = DefaultOrganizationForm()
+
+
+class DefaultTranslation(object):
+    def directory(self):
+        '''Change the directory of the *.mo translation files
+
+        The default implementation assumes the plugin is
+        ckanext/myplugin/plugin.py and the translations are stored in
+        i18n/
+        '''
+        # assume plugin is called ckanext.<myplugin>.<...>.PluginClass
+        extension_module_name = '.'.join(self.__module__.split('.')[0:2])
```

Comment:
@wardi, is this less offensive? I take the module name 'ckanext.whatever.blah.plugin' and assume the first two are 'ckanext.pluginname'. This feels a bit fragile.